Toolkit/ArcGISToolkit/AR/ArcGISARView.swift

```diff
@@ -0,0 +1,682 @@
+    }
+    
+    /// Handle a change in authorization status to "denied"
+    fileprivate func handleAuthStatusChangedAccessDenied() {
+        // auth status changed to denied
+        stopUpdatingLocationAndHeading()
+        
+        // we were waiting for user prompt to come back, so notify
+        didStartOrFailWithError(ArcGISARView.accessDeniedError())
+    }
+    
+    /// Handle a change in authorization status to "authorized"
+    fileprivate func handleAuthStatusChangedAccessAuthorized() {
+        // auth status changed to authorized
+        // we were waiting for status to come in to start the datasource
+        // now that we have authorization - start it
+        didStartOrFailWithError(nil)
+        
+        // need to start location manager updates
+        startUpdatingLocationAndHeading()
+    }
+    
+    /// Called when device orientation changes
+    ///
+    /// - Parameter notification: the notification
+    @objc func orientationChanged(notification: Notification?) {
+        // handle rotation here
+        switch UIApplication.shared.statusBarOrientation {
+        case .landscapeLeft:
+            orientationQuat = simd_quaternion(0, 0, 1.0, 0);
+        case .landscapeRight:
+            orientationQuat = simd_quaternion(0, 0, 0, 1.0);
+        case .portrait:
+            orientationQuat = simd_quaternion(0, 0, sqrt(0.5), sqrt(0.5));
+        case .portraitUpsideDown:
+            orientationQuat = simd_quaternion(0, 0, -sqrt(0.5), sqrt(0.5));
+        default:
+            break
+        }
+    }
+    
+    // MARK: Errors
+    
+    /// Error used when ARKit is not supported on the current device
+    ///
+    /// - Returns: error stating ARKit not supported
+    fileprivate class func notSupportedError() -> NSError {
+        let userInfo = [NSLocalizedDescriptionKey : "The device does not support ARKit functionality."]
+        return NSError(domain: AGSErrorDomain, code: 0, userInfo: userInfo)
+    }
+    
+    /// Error used when access to the device location is denied
+    ///
+    /// - Returns: error stating access to location information is denied
+    fileprivate class func accessDeniedError() -> NSError{
+        let userInfo = [NSLocalizedDescriptionKey : "Access to the device location is denied."]
+        return NSError(domain: kCLErrorDomain, code: CLError.Code.denied.rawValue, userInfo: userInfo)
+    }
+    
+    /// Error used when required plist information is missing
+    ///
+    /// - Returns: error stating plist information is missing
+    fileprivate class func missingPListKeyError() -> NSError{
+        let userInfo = [NSLocalizedDescriptionKey : "You must specify a location usage description key (NSLocationWhenInUseUsageDescription or NSLocationAlwaysUsageDescription) in your plist."]
+        return NSError(domain: kCLErrorDomain, code: CLError.Code.denied.rawValue, userInfo: userInfo)
+    }
+}
+
+// MARK: - ARSessionDelegate
+
+extension ArcGISARView: ARSessionDelegate {
```

Comment:
Is there a reason for them to be private?

Reply:
Good question. The answer is that it depends. There are at least two considerations:

1. Do you want the outside world to know that you conform to the protocol in question and which methods you implement?
2. Because the outside world knows about them, there is the potential that they could be called by someone other than the object of which you are the delegate.

I generally view conformances like this as implementation details and try to keep them internal. But the indirection does take extra work.

Reply:
I'm OK with having them public.